modules/mocked_authentication/spec/requests/credential_request_spec.rb

```diff
@@ -125,4 +125,132 @@
       end
     end
   end
+
+  describe 'GET credential_list' do
+    subject { get(credential_list_path, params: credential_list_params) }
+
+    let(:credential_list_path) { '/mocked_authentication/credential_list' }
+    let(:credential_list_params) { { type: csp_type } }
+    let(:csp_type) { 'logingov' }
+    let(:logingov_mock_data) do
+      path = File.join(
+        MockedAuthentication::Engine.root, 'spec',
+        'fixtures',
+        'credential_mock_data',
+        'logingov',
+        'mock_user_one.json'
+      )
+      JSON.parse(File.read(path))
+    end
+    let(:mock_user_one) { { user_identifier: 'vetsgovuser01', user_attributes: logingov_mock_data } }
+    let(:mock_user_two) { { user_identifier: 'vetsgovuser02', user_attributes: logingov_mock_data } }
+    let(:mock_user_three) { { user_identifier: 'vetsgovuser03', user_attributes: logingov_mock_data } }
+    let(:expected_mock_data) do
+      { mock_user_one[:user_identifier] =>
+          { 'credential_payload' => mock_user_one[:user_attributes],
+            'encoded_credential' => Base64.encode64(mock_user_one[:user_attributes].to_json) },
+        mock_user_two[:user_identifier] =>
+          { 'credential_payload' => mock_user_two[:user_attributes],
+            'encoded_credential' => Base64.encode64(mock_user_two[:user_attributes].to_json) },
+        mock_user_three[:user_identifier] =>
+          { 'credential_payload' => mock_user_three[:user_attributes],
+            'encoded_credential' => Base64.encode64(mock_user_three[:user_attributes].to_json) } }
+    end
+    let(:mock_data_directory) { "#{Settings.sign_in.mock_credential_dir}/credentials/#{csp_type}/" }
+    let(:mock_data_filenames) { expected_mock_data.keys.map { |k| "#{mock_data_directory}#{k}.json" } }
+
+    before do
+      allow(Dir).to receive(:glob).and_return(mock_data_filenames)
```

Comment:
I just thought of something, what if we mocked `Settings.sign_in.mock_credential_dir` to return the `fixtures/credential_mock_data` directory? It might simplify the mocking here by quite a bit.

The fixture would have to be moved into `fixtures/credential_mock_data/credentials/<csp_type>`